internal/services/automation/automation_variables_data_source.go

```diff
@@ -0,0 +1,289 @@
+package automation
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2020-01-13-preview/variable"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+func dataSourceAutomationVariables() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceAutomationVariablesRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"resource_group_name": commonschema.ResourceGroupName(),
+
+			"automation_account_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.RunbookName(),
+			},
```

Comment:
We can consolidate these two properties into one by using the ID

```suggestion
			"automation_account_id": {
				Type:         pluginsdk.TypeString,
				Required:     true,
				ValidateFunc: variable.ValidateAutomationAccountID,
			},
```